lib/lte_link_control/lte_lc_helpers.c

```diff
@@ -108,29 +125,110 @@ int event_handler_list_remove_handler(lte_lc_evt_handler_t handler)
 	return 0;
 }
 
+struct lte_lc_event_fifo_item *lte_lc_event_allocate(const struct lte_lc_evt *const evt)
+{
+	struct lte_lc_event_fifo_item *lte_lc_event_item;
+
+	lte_lc_event_item = k_heap_alloc(
+				&lte_lc_event_heap,
+				sizeof(struct lte_lc_event_fifo_item),
+				K_NO_WAIT);
+	if (!lte_lc_event_item) {
+		LOG_WRN("No LTE LC heap space for event type=%d", evt->type);
+		lte_lc_event_item = k_malloc(sizeof(struct lte_lc_event_fifo_item));
```

Comment:
I've made this allocation to provoke discussion also for `at_monitor` PR #15311.
If we do something like this. We can reduce the dedicated heap (lte_lc_event_heap) size a bit and use kernel heap (or even system heap) when available.

Reply:
Wouldn't you have to call `k_free()` for memory allocated with `k_malloc()` and `k_heap_free()` for memory allocated with `k_heap_alloc()`?

That said, I think this gets overly complicated. Are we trying to solve a real problem here? Do we have any numbers how many pending events there could be?

Reply:
> Wouldn't you have to call `k_free()` for memory allocated with `k_malloc()` and `k_heap_free()` for memory allocated with `k_heap_alloc()`?

Yep. Needs to be handled.

> That said, I think this gets overly complicated. Are we trying to solve a real problem here? Do we have any numbers how many pending events there could be?

Well, we either need to reserve extra memory for cases that may happen (and even then we are not sure if it's enough), or we will have fatal issues in form of an assert or dropping events.
I think we at least should have a discussion on something like this if we decide to dispatch lte_lc events from a dedicated workqueue.

Reply:
This was discussed in a meeting and we should not use multiple heaps for allocating memory.